src/docker-osx-dev

```diff
@@ -1031,6 +1034,23 @@ function install {
 }
 
 #
+# Fetch the latest docker-osx-dev script, and then run install
+#
+function update_self {
+  log_info "Updating to the latest docker-osx-dev"
+  local readonly current_file=$(which docker-osx-dev)
```

Comment:
It should be `${BASH_SOURCE[0]}` (note the placement of the braces). See this [SO discussion](http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in) for more details on the many ways a script can figure out its own path.

That said, now that I think about it further, this update approach has two potential problems:

1. A bash script that overwrites itself could lead to strange errors, as bash doesn't necessarily read the entire script into memory right at the start. See [this SO discussion](http://stackoverflow.com/questions/21096478/overwrite-executing-bash-script-files) for details.
2. This is a fairly flat upgrade scheme and doesn't include version numbers. I think that's OK for a first version, but to make debugging possible, we should probably think about having a version number in each install (e.g., the has of the latest git commit).